### PR TITLE
MRG, FIX: output_type for clustering

### LINKED ISSUE #8836: output of permutation_cluster_1samp_test seems wrong for out_type='indices'
Originally reported by @dimitriospantazis. Here is the code to reproduce (based on an MNE example):

```py
import numpy as np
import matplotlib.pyplot as plt

import mne
from mne.time_frequency import tfr_morlet
from mne.stats import permutation_cluster_1samp_test
from mne.datasets import sample

data_path = sample.data_path()
raw_fname = data_path + '/MEG/sample/sample_audvis_raw.fif'
tmin, tmax, event_id = -0.3, 0.6, 1

# Setup for reading the raw data
raw = mne.io.read_raw_fif(raw_fname)
events = mne.find_events(raw, stim_channel='STI 014')

include = []
raw.info['bads'] += ['MEG 2443', 'EEG 053']  # bads + 2 more

# picks MEG gradiometers
picks = mne.pick_types(raw.info, meg='grad', eeg=False, eog=True,
                       stim=False, include=include, exclude='bads')

# Load condition 1
event_id = 1
epochs = mne.Epochs(raw, events, event_id, tmin, tmax, picks=picks,
                    baseline=(None, 0), preload=True,
                    reject=dict(grad=4000e-13, eog=150e-6))
# just use right temporal sensors for speed
epochs.pick_channels(mne.read_selection('Right-temporal'))
evoked = epochs.average()

decim = 5
freqs = np.arange(8, 40, 2)  # define frequencies of interest
sfreq = raw.info['sfreq']  # sampling in Hz
tfr_epochs = tfr_morlet(epochs, freqs, n_cycles=4., decim=decim,
                        average=False, return_itc=False, n_jobs=1)

# Baseline power
tfr_epochs.apply_baseline(mode='logratio', baseline=(-.100, 0))

# Crop in time to keep only what is between 0 and 400 ms
evoked.crop(-0.1, 0.4)
tfr_epochs.crop(-0.1, 0.4)

epochs_power = tfr_epochs.data

threshold = 3.
n_permutations = 50  # Warning: 50 is way too small for real-world analysis.
data = epochs_power[0][0]
for out_type in ['mask', 'indices']:
    T_obs, clusters, cluster_p_values, H0 = \
        permutation_cluster_1samp_test(data, n_permutations=n_permutations,
                                    threshold=threshold, tail=1,
                                    adjacency=None,
                                    out_type=out_type, verbose=True)
    print(clusters)
```

I get:

```
Computing cluster p-values
Done.
[(slice(2, 4, None),), (slice(46, 49, None),)]
stat_fun(H1): min=-7.417671 max=4.199342
Running initial clustering
Found 2 clusters
Permuting 49 times...
100%|██████████████████████████████████████|  : 49/49 [00:00<00:00, 1025.40it/s]
Computing cluster p-values
Done.
[(array([0]),), (array([0]),)]
```

In the mask format, you get two different clusters but in the other format you get the same cluster repeated twice.

Looking at the [source code](https://github.com/mne-tools/mne-python/blob/4df9ed838f1d56471d071308fb62b9ec749cb10d/mne/stats/cluster_level) the loop does look buggy to me.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #8836

New output on that example is correct:
```
Running initial clustering
Found 2 clusters
Permuting 49 times...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|  : 49/49 [00:00<00:00, 6809.61it/s]
Computing cluster p-values
Done.
[(slice(2, 4, None),), (slice(46, 49, None),)]
stat_fun(H1): min=-7.417671 max=4.199342
Running initial clustering
Found 2 clusters
Permuting 49 times...
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████|  : 49/49 [00:00<00:00, 8677.99it/s]
Computing cluster p-values
Done.
[(array([2, 3]),), (array([46, 47, 48]),)]
```

### CODE REVIEW

File: mne/stats/cluster_level.py [hunk 1]
```diff
@@ -557,11 +557,20 @@ def _cluster_indices_to_mask(components, n_tot):
...
-        if not isinstance(c, slice):
-            components[ci] = np.where(c)[0]
+        if isinstance(c, np.ndarray):  # mask
+            components[ci] = np.where(c.reshape(shape))
+        else:
+            assert isinstance(c, tuple), type(c)
+            c = list(c)  # tuple->list
+            for ii, cc in enumerate(c):
+                if isinstance(cc, slice):
+                    c[ii] = np.arange(cc.start, cc.stop)
```
Comment (jasmainak) [inline mne/stats/cluster_level.py:570]: doesn't np.arange return a float? do you caste it to int somewhere?

Comment (larsoner) [reply mne/stats/cluster_level.py:570]: only if you use float arguments
```
>>> np.arange(1, 3)
array([1, 2])
>>> np.arange(1, 3).dtype
dtype('int64')
```
start and step here are int so it should be fine

**Review by jasmainak — APPROVED** [review]

**Comment by jasmainak** [comment]
I approved the PR but noticed an inconsistency afterwards. The docstring says:

> If 'indices', returns a list of lists, where each sublist contains the indices of locations that together form a cluster.

But in the example above, it's an array

**Review by drammock — APPROVED** [review]
really elegant test @larsoner.  Because of that I feel I needn't bother combing through the logic of the new code :)

**Pull request merged**